src/Spectre.libGaussianMixtureModelling/Matrix.h

```diff
@@ -66,6 +105,22 @@ struct Matrix
         return data[index];
     }
 
+    /// <summary>
+    /// Checks if two matrices are equal.
+    /// </summary>
+    /// <param name="other">Matrix to compare with.</param>
+    /// <param name="length">Width*height of the matrices</param>
+    /// <returns>True if matrices are equal, false otherwise</returns>
+    bool Equals(const Matrix& other, unsigned length)
+    {
+        for (unsigned i = 0; i < length; i++)
+        {
+            if (data[0][i] != other.data[0][i])
```

Comment:
We are using this for `float`s mostly. What about epsilon-based comparison?

Reply:
That was made just for the purpose of tests with indices (yet to come in Dyn Prog initialization)). I'd need to create a template specialized method for it.